src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -300,7 +369,8 @@ public void ExecuteSuppressionAction(DiagnosticSuppressor suppressor, ImmutableA
                 supportedSuppressions,
                 out Func<SuppressionDescriptor, bool> isSupportedSuppression);
 
-            var context = new SuppressionAnalysisContext(Compilation, AnalyzerOptions,
+            var options = GetAnalyzerSpecificOptions(suppressor);
```

Comment:
Putting a comment here, but this reminds me -- where is the code that computes effective severities for a diagnostic once produced? Do we also need to update that to respect the per diagnostic analyzer settings for that too?